pywren/executor.py

```diff
@@ -32,50 +36,38 @@ class Executor(object):
     Theoretically will allow for cross-AZ invocations
     """
 
-    def __init__(self, aws_region, s3_bucket, s3_prefix,
-                 invoker, runtime_s3_bucket, runtime_s3_key, job_max_runtime):
-        self.aws_region = aws_region
-        self.s3_bucket = s3_bucket
-        self.s3_prefix = s3_prefix
-
-        self.session = botocore.session.get_session()
+    def __init__(self, invoker, config, job_max_runtime):
         self.invoker = invoker
-        self.s3client = self.session.create_client('s3', region_name = aws_region)
         self.job_max_runtime = job_max_runtime
 
-        self.runtime_bucket = runtime_s3_bucket
-        self.runtime_key = runtime_s3_key
-        self.runtime_meta_info = runtime.get_runtime_info(runtime_s3_bucket, runtime_s3_key)
-        if not runtime.runtime_key_valid(self.runtime_meta_info):
-            raise Exception("The indicated runtime: s3://{}/{} is not approprite for this python version".format(runtime_s3_bucket, runtime_s3_key))
+        self.config = config
+        self.storage_config = wrenconfig.extract_storage_config(self.config)
```

Comment:
Can we formalize what `config` looks like? Also, it might be easier to have just one `self.config` instead of both `self.storage_config` and `self.config`